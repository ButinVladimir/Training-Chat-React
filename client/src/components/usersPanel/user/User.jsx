import React from 'react';
import PropTypes from 'prop-types';

function User({ user, onClickUser }) {
  return (
    <li key={user} className="user-line">
      <button
        type="button"
        className="user-select"
        onClick={onClickUser}
      >
        {user}
      </button>
    </li>
  );
}

User.propTypes = {
  user: PropTypes.string.isRequired,
  onClickUser: PropTypes.func.isRequired,
};

export default User;
