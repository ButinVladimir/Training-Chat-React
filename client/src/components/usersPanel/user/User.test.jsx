import React from 'react';
import { shallow } from 'enzyme';
import User from './User';

describe('User', () => {
  it('renders without crashing', () => {
    const component = shallow(<User user="user" onClickUser={() => {}} />);
    expect(component).toMatchSnapshot();
  });

  it('handles click on user', () => {
    const mock = jest.fn();
    const component = shallow(<User user="user" onClickUser={mock} />);

    component.find('.user-select').simulate('click');

    expect(mock).toHaveBeenCalledTimes(1);
    expect(mock).toHaveBeenCalledWith('user');
  });
});
