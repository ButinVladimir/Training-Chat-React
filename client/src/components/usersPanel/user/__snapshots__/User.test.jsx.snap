// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User renders without crashing 1`] = `
<li
  className="user-line"
  key="user"
>
  <button
    className="user-select"
    onClick={[Function]}
    type="button"
  >
    user
  </button>
</li>
`;
